src/test/java/core/basesyntax/OperationsStrategyImplTest.java

```diff
@@ -0,0 +1,111 @@
+import java.util.EnumMap;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class OperationsStrategyImplTest {
+    private static OperationsStrategyImpl strategy;
+    private static Map<Operations, OperationHandler> operationHandlerMap;
+
+    @BeforeEach
+    public void setUp() {
+        operationHandlerMap = new EnumMap<>(Operations.class);
+        operationHandlerMap.put(Operations.BALANCE, new BalanceOperationsHandler());
+        operationHandlerMap.put(Operations.SUPPLY, new SupplyOperationsHandler());
+        operationHandlerMap.put(Operations.PURCHASE, new PurchaseOperationsHandler());
+        operationHandlerMap.put(Operations.RETURN, new ReturnOperationHandler());
+        strategy = new OperationsStrategyImpl(operationHandlerMap);
+    }
+
+    @Test
+    void get_exceptionNull_NotOk() {
+        assertThrows(UnsupportedOperationException.class, () -> {
+            OperationHandler handler = strategy.get(null);
+            handler.getOperation();
+        });
+    }
+
+    @Test
+    void get_supplyHandlerBySupplyOperation_Ok() {
+        OperationHandler handler = strategy.get(Operations.SUPPLY);
+        assertEquals(SupplyOperationsHandler.class, handler.getClass());
+    }
+
+    @Test
+    void get_purchaseHandlerByPurchaseOperation_Ok() {
+        OperationHandler handler = strategy.get(Operations.PURCHASE);
+        assertEquals(PurchaseOperationsHandler.class, handler.getClass());
+    }
+
+    @Test
+    void get_balanceHandlerByBalanceOperation_Ok() {
+        OperationHandler handler = strategy.get(Operations.BALANCE);
+        assertEquals(BalanceOperationsHandler.class, handler.getClass());
+    }
+
+    @Test
+    void get_returnHandlerByBalanceOperation_Ok() {
+        OperationHandler handler = strategy.get(Operations.RETURN);
+        assertEquals(ReturnOperationHandler.class, handler.getClass());
+    }
+
+    @Test
+    void get_operationSypplyFromSypplyHandler_Ok() {
+        SupplyOperationsHandler handler = new SupplyOperationsHandler();
+        assertEquals(Operations.SUPPLY, handler.getOperation());
+    }
+
+    @Test
+    void get_operationPurchaseFromPurchaseHandler_Ok() {
+        PurchaseOperationsHandler handler = new PurchaseOperationsHandler();
+        assertEquals(Operations.PURCHASE, handler.getOperation());
+    }
+
+    @Test
+    void get_operationBalanceFromBalanceHandler_Ok() {
+        BalanceOperationsHandler handler = new BalanceOperationsHandler();
+        assertEquals(Operations.BALANCE, handler.getOperation());
+    }
+
+    @Test
+    void get_operationReturnFromReturnHandler_Ok() {
+        ReturnOperationHandler handler = new ReturnOperationHandler();
+        assertEquals(Operations.RETURN, handler.getOperation());
+    }
+
+    @Test
+    void apply_operationFromSupplyHandler_Ok() {
+        SupplyOperationsHandler handler = new SupplyOperationsHandler();
+        assertEquals(11, handler.applyOperation(7, 4));
```

Comment:
this class is not a correct place to test SupplyOperationsHandler. Fix all other cases as well